services/streammanager/kafka/kafkamanager_test.go

```diff
@@ -214,7 +214,7 @@ func TestNewProducerForAzureEventHubs(t *testing.T) {
 
 		require.Eventually(t, func() bool {
 			ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
-			err = p.p.Publish(ctx, client.Message{Key: []byte("key-01"), Value: []byte("message-01")})
+			err = p.p.Publish(ctx, client.Message{Key: []byte("key-01"), Value: []byte("message-01"), Topic: destConfig["topic"].(string)})
```

Comment:
@utsabc, I saw the changes and could see you have used a single-letter variable name for the `Producer` struct. Generally, you can try to use more meaningful names. Single var names are okay if you are creating a library but if this logic will be worked upon in future also then it is very difficult to read something like `p.p` when compared to `someProducer.produce` or `someProducer.childProducer`. (I couldn't understand what the second p is doing).

Reply:
Yeah this was done as part of original implementation, I didn't want to make refactor there as it was not part of the original scope. But I see your point

Reply:
I wrote the original implementation this way because I find the pattern common in Go codebases where one struct is just a wrapper of another so the second `p` is just the internal producer.

If it's not readable for you we can rename it in another PR to something like `p.client` or `p.internalClient`.